test/k8sT/Updates.go

```diff
@@ -108,15 +113,16 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
 		// make sure that Kubedns is deleted correctly
 		_ = kubectl.Delete(helpers.DNSDeployment())
 
+		kubectl.DeleteETCDOperator()
+
 		// make sure we clean everything up before doing any other test
 		err := kubectl.CiliumInstall(
 			helpers.CiliumDefaultDSPatch,
 			"cilium-cm-patch-clean-cilium-state.yaml",
 		)
 		ExpectWithOffset(1, err).To(BeNil(), "Cilium %q was not able to be deployed", newVersion)
-
-		ExpectAllPodsTerminated(kubectl)
-		ExpectCiliumReady(kubectl)
+		err = kubectl.WaitForCiliumInitContainerToFinish()
+		ExpectWithOffset(1, err).To(BeNil(), "Cilium %q was not able to be clean up environment", newVersion)
```

Comment:
This should be moved at the end of the file. This cleanup happens on the AfterEAch and the asserts on AfterEach are not working well. Should happen at the end.

Reply:
Can you clarify what you mean?

Reply:
Move from cleanup function to the original one at the end. The cleanup happens on the AfterEach.